test/functional/wallet_orphanedreward.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test orphaned block rewards in the wallet."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, assert_raises_rpc_error
+
+class OrphanedBlockRewardTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
```

Comment:
I think that a clean chain makes the test easier to read, as it doesn't require any prior knowledge about how the cached chain looks like.  If we use the cached chain, it will be harder to make sure that `node[1]` only has the coins we want to, as it will have immature coins that will appear over time.

Also, it seems the initial setup (including until after the `sendtoaddress`) is quick anyway.  The slow parts are the `sync_blocks` calls, which we wouldn't get rid of with a cached chain.

Or did I misunderstand your suggestion?